Another hr button fix

w-e-w opened the pull request:
## Description

use `force_enable_hr` to set `p.enable_hr = True` in `txt2img_create_processing`
allow `Script.setup()` have access to the correct value

add a comment for p.txt2img_upscale

<details><summary>old</summary>
<p>

## Description

1. [rework set_named_arg](https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/118a2273ed63e7eaedd399bbd5189512da7488f2)

- the identifying a script using `class name` is problematic as currently as there's no convention to keep unique names with there `script class name`
in base webui alone we have multiple scripts that are all called Scripts
not to mention lots of extensions may use the same Script name for the Script
change to using the scrip internal `name`

![2024-01-22 18_51_30_697](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/40751091/30a032b7-7d48-4827-9db6-5460a34e6441)
> the first tree is scripts from web UI that are just named Scripts, 

- and as opposed to returning none when there is an issue like script or element id is not found, raise runtime error

- even though comparing the element ID using `>` is convenient as it allows you to ignore stuff like the prefix `txt2img_` in `txt2img_seed` but this would cause a potential issue in that depending on the order of your arg
for example if you have two elements args with IDs of `seed` and `subseed`
if the order of the args is `['seed', 'subseed']` theneverything function as normal
but if the order is `['subseed', 'seed']` when matching for `seed` it be match with `subseed`
hence changed to using `==`

2. adding two new helper functions for extracting script args and setting script args
- - [update_script_args](https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/f0ef4e8b6ce74713f007c72af837131f421f88f4) functions similar to `set_named_arg` but for the entire args
- - [get_script_args](https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/6b07a935dfd15acc7822e3fd596dadef2994f78c)

3. [rework `txt2img_upscale` (Hr button) p create argument](https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/96d6e3e76bfa114379857e2f82483ef19719ccc6)
- - update all the arguments before p is actually created so that `Script.setup()` works with correct args
- - as it is possible for a certain extensions to utilize `enable_hr` `batch_size` ... in there `Script.setup()`
- - - I was using `enable_hr` in `Script.setup()` of my extension, and noticed that the value is not set properly

- update the entire seed script arg, previously only seed and subseed were read from infotext, now everything is form about seed infotext
- - this means when using Hr button, the `seed UI` is completely ignored

</p>
</details> 

## Checklist:

- [x] I have read [contributing wiki page](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Contributing)
- [x] I have performed a self-review of my own code
- [x] My code follows the [style guidelines](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Contributing#code-style)
- [x] My code passes [tests](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Tests)

AUTOMATIC1111 commented:
I kind of don't like the more complicated implementation for txt2img_upscale and I don't understand why it's needed.

The change from script classname to script name is fine but I'd rather it was a single PR.

And as for addition of two other functions for settings script args, I'm not sure if they are needed too.

w-e-w commented:
theres also an issue with `set_named_arg`
the arg you want to set has to have a pre-defined arg_elem_id

AUTOMATIC1111 commented:
> this I believe is implemented what you initially tried to implement then accidentally broke irrc you initially try to set the seat using the set_named_arg you added for that specific purpose but then you implemented something else which means the setup is already executed so I implemented stuff to do every argument setting before it even creates `p`
> 
> the main point of this is that this makes `setup()` behave exactly the same as normal txt2img otherwise there will be slight differences that might be important for certain extensions, yes those extension most likely can adapt but it can also cause some potential bugs that is not going to be caught after a long time
> 
> I see this as slightly complicated it's here but reduce complications elsewhere

Well. We create `p`. It works just as usual. Setup runs just as usual. It sets the seed. Later in the program we change this seed. This is not abnormal. Custom scripts that create their own `p` will also do this (like XYZ plot). Why the need to touch the arguments set to `p`?

AUTOMATIC1111 commented:
I kind of don't want to do positional script args. If those functions works with a dictionary of name/value pairs, that would be nice.

w-e-w committed:
fix txt2img_upscale

use force_enable_hr to set p.enable_hr = True
allow Script.setup() have access to the correct value

add a comment for p.txt2img_upscale

w-e-w commented:
> The change from script classname to script name is fine but I'd rather it was a single PR.

moved to 

- https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/14773

- remove everything else that is not strictly necessary ,leaving only `force_enable_hr`

> For txt2img_upscale, yes, unless there's a reason to do this your way, I'd like to keep it simpler.
> update_script_args get_script_args: I kind of don't want to do positional script args. If those functions works with a dictionary of name/value pairs, that would be nice.

removed for now
